express/components/common.py

```diff
@@ -341,10 +391,17 @@ def run(cls) -> DataManifest:
         input_dataset = cls._load_dataset(
             input_manifest=DataManifest.from_path(args.input_manifest)
         )
+        # verify incoming data sources
+        cls._verify_data_sources_in(input_dataset)
```

Comment:
I would like to see functionality to load a dataset based on a manifest and component specification instead. Where the component specification is used to decide which modalities and columns need to be loaded from the manifest, and the manifest is used to get the path to the actual data.